tket/src/Circuit/macro_circ_info.cpp

```diff
@@ -708,6 +708,21 @@ unsigned Circuit::depth_by_types(const OpTypeSet& _types) const {
   return count;
 }
 
+unsigned Circuit::depth_by_2q_gates() const {
+  unsigned count = 0;
+  std::function<bool(Op_ptr)> skip_func = [&](Op_ptr op) {
+    return (op->n_qubits() != 2 || !is_gate_type(op->get_type()));
```

Comment:
Is `!is_gate_type(op->get_type()));` excluding conditionals and boxes or should they work fine?

Reply:
Yes, boxes and conditionals are not counted, but I'm not certain about this. The existing `depth_by_type` and `depth_by_types` methods don't count the conditional version of the given gate types, so I'm not sure if I should keep this behavior.

I named this method `depth_by_2q_gates` with the intention of not counting boxes. I'm not sure if boxes are considered "gates" since they can contain anything (e.g., measurements, conditionals, barriers).

My rationale is that people want a method that is equivalent to passing all 2-qubit gate types to `depth_by_type`, and this method achieves that while also recognising gates such as CnX.

Perhaps I can name this method something like `depth_by_2q_vertices` and count every vertex that has 2 qubits, as it might be less confusing?

I'm open to suggestions.

Reply:
I would have expected that boxes and conditionals that only interact with two qubits gates are counted as well. But I am not sure if that is what other people would expect.

Is this implemented based on a user request?

Maybe we can check this with them or add both in two functions and point out the differences in the documentation?

I think my preferred solution (if there is not further feedback) would be to include conditionals and boxes (and maybe rename the function)